<Type Name="ArithmeticException" FullName="System.ArithmeticException">
  <Metadata><Meta Name="ms.openlocfilehash" Value="d4091774aeb94d964164e19f7b72854106fd63b7" /><Meta Name="ms.sourcegitcommit" Value="c0c07dbd19cd7017243f9ac36915755f79bc8da6" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="pt-BR" /><Meta Name="ms.lasthandoff" Value="11/27/2018" /><Meta Name="ms.locfileid" Value="52362313" /></Metadata><TypeSignature Language="C#" Value="public class ArithmeticException : SystemException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit ArithmeticException extends System.SystemException" />
  <TypeSignature Language="DocId" Value="T:System.ArithmeticException" />
  <TypeSignature Language="VB.NET" Value="Public Class ArithmeticException&#xA;Inherits SystemException" />
  <TypeSignature Language="C++ CLI" Value="public ref class ArithmeticException : SystemException" />
  <TypeSignature Language="F#" Value="type ArithmeticException = class&#xA;    inherit SystemException" />
  <AssemblyInfo>
    <AssemblyName>System.Runtime</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
    <AssemblyVersion>4.0.10.0</AssemblyVersion>
    <AssemblyVersion>4.0.20.0</AssemblyVersion>
    <AssemblyVersion>4.1.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>1.0.5000.0</AssemblyVersion>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Exception</BaseTypeName>
    <BaseTypeName FrameworkAlternate="netframework-4.5">System.SystemException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute FrameworkAlternate="netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0;netframework-4.8">
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
    <Attribute FrameworkAlternate="netframework-1.1;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0">
      <AttributeName>System.Serializable</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary><span data-ttu-id="f1555-101">A exceção que é gerada para erros em uma operação aritmética, projeção ou conversão.</span><span class="sxs-lookup"><span data-stu-id="f1555-101">The exception that is thrown for errors in an arithmetic, casting, or conversion operation.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f1555-102"><xref:System.ArithmeticException> é a classe base para as seguintes exceções:</span><span class="sxs-lookup"><span data-stu-id="f1555-102"><xref:System.ArithmeticException> is the base class for the following exceptions:</span></span>  
  
-   <span data-ttu-id="f1555-103"><xref:System.DivideByZeroException>, que é gerado na divisão de inteiro quando o divisor for 0.</span><span class="sxs-lookup"><span data-stu-id="f1555-103"><xref:System.DivideByZeroException>, which is thrown in integer division when the divisor is 0.</span></span>  <span data-ttu-id="f1555-104">Por exemplo, a tentativa de dividir 10 por 0 gera uma <xref:System.DivideByZeroException> exceção.</span><span class="sxs-lookup"><span data-stu-id="f1555-104">For example, attempting to divide 10 by 0 throws a <xref:System.DivideByZeroException> exception.</span></span>  
  
-   <span data-ttu-id="f1555-105"><xref:System.NotFiniteNumberException>, que é gerado quando uma operação é executada em ou retorna <xref:System.Double.NaN?displayProperty=nameWithType>, <xref:System.Double.NegativeInfinity?displayProperty=nameWithType>, <xref:System.Double.PositiveInfinity?displayProperty=nameWithType>, <xref:System.Single.NaN?displayProperty=nameWithType>, <xref:System.Single.NegativeInfinity?displayProperty=nameWithType>, <xref:System.Single.PositiveInfinity?displayProperty=nameWithType>, e a linguagem de programação usada não dá suporte a esses valores.</span><span class="sxs-lookup"><span data-stu-id="f1555-105"><xref:System.NotFiniteNumberException>, which is thrown when an operation is performed on or returns <xref:System.Double.NaN?displayProperty=nameWithType>, <xref:System.Double.NegativeInfinity?displayProperty=nameWithType>, <xref:System.Double.PositiveInfinity?displayProperty=nameWithType>, <xref:System.Single.NaN?displayProperty=nameWithType>, <xref:System.Single.NegativeInfinity?displayProperty=nameWithType>, <xref:System.Single.PositiveInfinity?displayProperty=nameWithType>, and the programming language used does not support those values.</span></span>  
  
-   <span data-ttu-id="f1555-106"><xref:System.OverflowException>, que é gerado quando o resultado de uma operação está fora dos limites do tipo de dados de destino.</span><span class="sxs-lookup"><span data-stu-id="f1555-106"><xref:System.OverflowException>, which is thrown when the result of an operation is outside the bounds of the target data type.</span></span> <span data-ttu-id="f1555-107">Ou seja, ele é menor que um número `MinValue` propriedade ou maior que sua `MaxValue` propriedade.</span><span class="sxs-lookup"><span data-stu-id="f1555-107">That is, it is less than a number's `MinValue` property or greater than its `MaxValue` property.</span></span>  <span data-ttu-id="f1555-108">Por exemplo, a tentativa de atribuir 200 + 200 para um <xref:System.Byte> valor lança uma <xref:System.OverflowException> exceção desde 400 maior que 256, o limite superior do <xref:System.Byte> tipo de dados.</span><span class="sxs-lookup"><span data-stu-id="f1555-108">For example, attempting to assign 200 + 200 to a <xref:System.Byte> value throws an <xref:System.OverflowException> exception, since 400 greater than 256, the upper bound of the <xref:System.Byte> data type.</span></span>  
  
 <span data-ttu-id="f1555-109">Seu código não deve tratar ou lançar essa exceção.</span><span class="sxs-lookup"><span data-stu-id="f1555-109">Your code should not handle or throw this exception.</span></span> <span data-ttu-id="f1555-110">Em vez disso, você deve tratar ou lançar uma de suas classes derivadas, pois indica a natureza exata do erro com mais precisão.</span><span class="sxs-lookup"><span data-stu-id="f1555-110">Instead, you should either handle or throw one of its derived classes, since it more precisely indicates the exact nature of the error.</span></span>  
  
 <span data-ttu-id="f1555-111">Para obter uma lista de valores de propriedade inicial para uma instância do <xref:System.ArithmeticException>, consulte o <xref:System.ArithmeticException> construtores.</span><span class="sxs-lookup"><span data-stu-id="f1555-111">For a list of initial property values for an instance of <xref:System.ArithmeticException>, see the <xref:System.ArithmeticException> constructors.</span></span>  
  
 <span data-ttu-id="f1555-112"><xref:System.ArithmeticException> usa o HRESULT COR_E_ARITHMETIC, que tem o valor 0x80070216.</span><span class="sxs-lookup"><span data-stu-id="f1555-112"><xref:System.ArithmeticException> uses the HRESULT COR_E_ARITHMETIC, which has the value 0x80070216.</span></span>  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Exception" />
    <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="f1555-113">Tratando e gerando exceções no .NET</span><span class="sxs-lookup"><span data-stu-id="f1555-113">Handling and throwing exceptions in .NET</span></span></related>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="f1555-114">Inicializa uma nova instância da classe <see cref="T:System.ArithmeticException" />.</span><span class="sxs-lookup"><span data-stu-id="f1555-114">Initializes a new instance of the <see cref="T:System.ArithmeticException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ArithmeticException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.ArithmeticException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ArithmeticException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="f1555-115">Inicializa uma nova instância da classe <see cref="T:System.ArithmeticException" />.</span><span class="sxs-lookup"><span data-stu-id="f1555-115">Initializes a new instance of the <see cref="T:System.ArithmeticException" /> class.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f1555-116">Este construtor inicializa o <xref:System.Exception.Message%2A> propriedade da nova instância para uma mensagem fornecida pelo sistema que descreve o erro, por exemplo, "a operação aritmética não é permitida."</span><span class="sxs-lookup"><span data-stu-id="f1555-116">This constructor initializes the <xref:System.Exception.Message%2A> property of the new instance to a system-supplied message that describes the error, such as "The arithmetic operation is not allowed."</span></span> <span data-ttu-id="f1555-117">Esta mensagem leva em conta a cultura do sistema atual.</span><span class="sxs-lookup"><span data-stu-id="f1555-117">This message takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="f1555-118">A tabela a seguir mostra os valores de propriedade inicial para uma instância do <xref:System.ArithmeticException>.</span><span class="sxs-lookup"><span data-stu-id="f1555-118">The following table shows initial property values for an instance of <xref:System.ArithmeticException>.</span></span>  
  
|<span data-ttu-id="f1555-119">Propriedade</span><span class="sxs-lookup"><span data-stu-id="f1555-119">Property</span></span>|<span data-ttu-id="f1555-120">Valor</span><span class="sxs-lookup"><span data-stu-id="f1555-120">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="f1555-121">`null`.</span><span class="sxs-lookup"><span data-stu-id="f1555-121">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="f1555-122">A cadeia de caracteres da mensagem de erro localizada.</span><span class="sxs-lookup"><span data-stu-id="f1555-122">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ArithmeticException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.ArithmeticException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ArithmeticException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new ArithmeticException : string -&gt; ArithmeticException" Usage="new System.ArithmeticException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="f1555-123">Um <see cref="T:System.String" /> que descreve o erro.</span><span class="sxs-lookup"><span data-stu-id="f1555-123">A <see cref="T:System.String" /> that describes the error.</span></span></param>
        <summary><span data-ttu-id="f1555-124">Inicializa uma nova instância da classe <see cref="T:System.ArithmeticException" /> com uma mensagem de erro especificada.</span><span class="sxs-lookup"><span data-stu-id="f1555-124">Initializes a new instance of the <see cref="T:System.ArithmeticException" /> class with a specified error message.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f1555-125">O conteúdo do parâmetro `message` deve ser compreendido por humanos.</span><span class="sxs-lookup"><span data-stu-id="f1555-125">The content of the `message` parameter is intended to be understood by humans.</span></span> <span data-ttu-id="f1555-126">O chamador desse construtor é necessário para garantir que essa cadeia de caracteres tenha sido localizada para a cultura do sistema atual.</span><span class="sxs-lookup"><span data-stu-id="f1555-126">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span></span>  
  
 <span data-ttu-id="f1555-127">A tabela a seguir mostra os valores de propriedade inicial para uma instância do <xref:System.ArithmeticException>.</span><span class="sxs-lookup"><span data-stu-id="f1555-127">The following table shows initial property values for an instance of <xref:System.ArithmeticException>.</span></span>  
  
|<span data-ttu-id="f1555-128">Propriedade</span><span class="sxs-lookup"><span data-stu-id="f1555-128">Property</span></span>|<span data-ttu-id="f1555-129">Valor</span><span class="sxs-lookup"><span data-stu-id="f1555-129">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="f1555-130">A referência nula (`Nothing` no Visual Basic).</span><span class="sxs-lookup"><span data-stu-id="f1555-130">The null reference (`Nothing` in Visual Basic).</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="f1555-131">A cadeia de caracteres da mensagem de erro.</span><span class="sxs-lookup"><span data-stu-id="f1555-131">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected ArithmeticException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.ArithmeticException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; ArithmeticException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new ArithmeticException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; ArithmeticException" Usage="new System.ArithmeticException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Security.SecuritySafeCritical</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="f1555-132">O objeto que mantém os dados de objeto serializados.</span><span class="sxs-lookup"><span data-stu-id="f1555-132">The object that holds the serialized object data.</span></span></param>
        <param name="context"><span data-ttu-id="f1555-133">As informações contextuais sobre a origem ou o destino.</span><span class="sxs-lookup"><span data-stu-id="f1555-133">The contextual information about the source or destination.</span></span></param>
        <summary><span data-ttu-id="f1555-134">Inicializa uma nova instância da classe <see cref="T:System.ArithmeticException" /> com dados serializados.</span><span class="sxs-lookup"><span data-stu-id="f1555-134">Initializes a new instance of the <see cref="T:System.ArithmeticException" /> class with serialized data.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f1555-135">Este construtor é chamado durante a desserialização para reconstituir o objeto de exceção transmitido por um fluxo.</span><span class="sxs-lookup"><span data-stu-id="f1555-135">This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</span></span> <span data-ttu-id="f1555-136">Para obter mais informações, consulte [serialização XML e SOAP](~/docs/standard/serialization/xml-and-soap-serialization.md).</span><span class="sxs-lookup"><span data-stu-id="f1555-136">For more information, see [XML and SOAP Serialization](~/docs/standard/serialization/xml-and-soap-serialization.md).</span></span>  
  
 ]]></format>
        </remarks>
        <related type="Article" href="~/docs/standard/serialization/xml-and-soap-serialization.md"><span data-ttu-id="f1555-137">Serialização XML e SOAP</span><span class="sxs-lookup"><span data-stu-id="f1555-137">XML and SOAP Serialization</span></span></related>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ArithmeticException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.ArithmeticException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ArithmeticException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new ArithmeticException : string * Exception -&gt; ArithmeticException" Usage="new System.ArithmeticException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="f1555-138">A mensagem de erro que explica a razão da exceção.</span><span class="sxs-lookup"><span data-stu-id="f1555-138">The error message that explains the reason for the exception.</span></span></param>
        <param name="innerException"><span data-ttu-id="f1555-139">A exceção que é a causa da exceção atual.</span><span class="sxs-lookup"><span data-stu-id="f1555-139">The exception that is the cause of the current exception.</span></span> <span data-ttu-id="f1555-140">Se o parâmetro <paramref name="innerException" /> não for uma referência nula, a exceção atual será gerada em um bloco <see langword="catch" /> que lida com a exceção interna.</span><span class="sxs-lookup"><span data-stu-id="f1555-140">If the <paramref name="innerException" /> parameter is not a null reference, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span></span></param>
        <summary><span data-ttu-id="f1555-141">Inicializa uma nova instância da classe <see cref="T:System.ArithmeticException" /> com uma mensagem de erro especificada e uma referência à exceção interna que é a causa da exceção.</span><span class="sxs-lookup"><span data-stu-id="f1555-141">Initializes a new instance of the <see cref="T:System.ArithmeticException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f1555-142">Uma exceção que é lançada como um resultado direto de uma exceção anterior deve incluir uma referência para a exceção anterior na propriedade <xref:System.Exception.InnerException%2A>.</span><span class="sxs-lookup"><span data-stu-id="f1555-142">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="f1555-143">A propriedade <xref:System.Exception.InnerException%2A> retorna o mesmo valor que é passado para o construtor ou uma referência nula (`Nothing` no Visual Basic) caso a propriedade <xref:System.Exception.InnerException%2A> não forneça o valor de exceção interna para o construtor.</span><span class="sxs-lookup"><span data-stu-id="f1555-143">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or a null reference (`Nothing` in Visual Basic) if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="f1555-144">A tabela a seguir mostra os valores de propriedade inicial para uma instância do <xref:System.ArithmeticException>.</span><span class="sxs-lookup"><span data-stu-id="f1555-144">The following table shows initial property values for an instance of <xref:System.ArithmeticException>.</span></span>  
  
|<span data-ttu-id="f1555-145">Propriedade</span><span class="sxs-lookup"><span data-stu-id="f1555-145">Property</span></span>|<span data-ttu-id="f1555-146">Valor</span><span class="sxs-lookup"><span data-stu-id="f1555-146">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="f1555-147">A referência de exceção interna.</span><span class="sxs-lookup"><span data-stu-id="f1555-147">The inner exception reference.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="f1555-148">A cadeia de caracteres da mensagem de erro.</span><span class="sxs-lookup"><span data-stu-id="f1555-148">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Exception" />
        <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="f1555-149">Manipulando e lançando exceções</span><span class="sxs-lookup"><span data-stu-id="f1555-149">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
  </Members>
</Type>